processing/src/main/java/org/apache/druid/frame/channel/PartitionedReadableFrameChannel.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.frame.channel;
+
+import java.io.Closeable;
+
+/**
+ * Provides an interface to read a partitioned frame channel. The channel might have frames with multiple partitions
+ * in it.
+ */
+public interface PartitionedReadableFrameChannel extends Closeable
```

Comment:
This is more of a supplier than an actual channel, so `PartitionedReadableFrameChannelSupplier`, or something like that, would be a clearer name.

Javadocs should also describe what happens when this is closed.

Reply:
I tried changing it to `PartitionedReadableFrameChannelSupplier` but it becomes a bit confusing later on since the readable channels are sent around wrapped in another `Supplier` object. It ends up looking like `Supplier< PartitionedReadableFrameChannelSupplier>`.
I have changed the method name a bit in the interface to make things more clear. Please let me know if this looks ok or should we change to `supplier` suffix for more clarity